### ISSUE TITLE
Bug: camera issues when exceeding screen size

### ISSUE THREAD

**Issue by rafcolson** [GoranM/bdx#628]
Two issues:

1. There's an inconsistency between `Orthographic` and `Perspective` camera when the Blender Render Resolution exceeds the screen size. For example when the Blender Render Resolution is 1920 X 1080 and the screen size is 1366 X 768 `Orthographic` will fit to the width, while `Perspective` will fit to the height. This will especially be noticeable with square resolutions, for example 1024 X 1024.

2. Right now the `Camera.width()` and `Camera.height()` are copied from `Bdx.display.size()`. The problem with this is that `Gdx.graphics.getWidth()` and `Gdx.graphics.getWidth()` rely on the width and height of the screen. For example, if the screen resolution is 1366 X 768 and Blender Resolution is 1366 X 768, then the width and height will be 1360 X 735. This is due to the fact that BDX starts out with a windowed screen, which has a border and a top bar (this applies for Windows, don't know if it's the same with OSX or Linux). For this reason adjusting the width and height to get the right aspect ratio on different screens is troublesome. If I want to play my 1280 X 768 game on a 1366 X 768 screen while preserving aspect ratio I'm not sure I can, whereas if the camera width and height starts out with the actual Blender Render Resolution (in a test commit), everything shows up fine after changing the camera width and height to the full screen display size.
```java
 		Bdx.display.fullscreen(true);
		Vector2f ds = Bdx.display.size();
		scene.camera.width(ds.x);
		scene.camera.height(ds.y);
```
I think we should change this and copy from the Blender's Render tab's Resolution instead, so that the initial resolution of the game screen is preserved. This should also solve the inconsistency between `Orthographic` and `Perspective`.

Edit: Additionally, we could also utilize Extend, Letterbox and Scale modes so the scene's camera width and height are updated with Extend and Letterbox when the display size changes. Right now the camera always scales.

Edit: Resizing the camera seems tedious. Can't get it right. So I removed the example.

**Comment by GoranM** [comment]
How would resolution exceed screen size? Shouldn't the screen size be defined by the resolution?

**Comment by rafcolson** [comment]
What I mean is that when the Blender Resolution in the Render tab(resolution) exceeds the resolution of your computer or tablet screen (screen size), this causes some issues, as described in my opening post.

I have a fix but I guess you and @SolarLune have to see what you think about the changes. The main change is that the windowed BDX screen now can exceed the borders of your computer screen, meaning that viewport dimensions `Camera.width()` and `Camera.height()` are independent of `Bdx.display.size()`. So now we can chose between Extend or Scale Framing. Letterbox could work when Viewports are in, I suppose.

**Comment by GoranM** [comment]
Oh, I see.

Your suggested change seems reasonable to me.

**Comment by rafcolson** [comment]
PR submitted.

**Comment by SolarLune** [comment]
This is cool, but I'm not sure how this will work with viewports, as they already deal with resolution changes and scaling modes. Like, if we implement this, and then go to implement viewports and the LibGDX ones can't be used for whatever reason, would it be OK to implement our own viewport solution, or would we roll this back?

**Comment by rafcolson** [comment]
>This is cool, but I'm not sure how this will work with viewports, as they already deal with resolution changes and scaling modes.

As far as I know Viewports don't deal with EXTEND and FIXED modes, except for SCALE and LETTERBOX. But you go first. That's probably better. There may be some rolling back otherwise.

So I'll wait until Viewports are in.

**Comment by SolarLune** [comment]
Eh, feel free to go ahead; not sure when I'll feel like getting back to doing Viewports.

**Comment by rafcolson** [comment]
Okay. I'll give it a try.

**Comment by SolarLune** [comment]
Hey, I think I'm seeing this bug. 

When I start my game and maximize the window to fill the screen, everything appears normally. However, when I restart it with the window large, part of the top of the screen is cut off (maybe because BDX thinks the window's resolution's 1920x1080, without taking into account the top section). I'm using Linux, Ubuntu Gnome 15.10. 

Is this what you were referring to, @rafcolson?

**Comment by rafcolson** [comment]
If with 'top section' you mean 'top bar', then, yes. A quick fix is to export scene resolution to set the camera width and height at initialization.

**Comment by rafcolson** [comment]
This now is solved by #713.
